app/controllers/application_controller.rb

```diff
@@ -172,9 +172,27 @@ def forward_to_app_config_domain
     redirect_to URL.url(request.fullpath)
   end
 
+  def current_user
+    return super unless edge_cached_action?
+
+    return nil unless session_current_user_id.present?
+      message = "😱😱😱😱😱 You cannot use current_user in this action\
+  because this code path will be cached at the edge in production.\
+  Check out this section in the docs: https://docs.forem.com/technical-overview/architecture/#we-cache-many-content-pages-on-the-edge"
+      p message
+      Object.const_set("YouCannotUseCurrentUserHereCheckTheTerminalOutput", Class.new)
```

Comment:
1. What is the purpose of this?
2. Classes really shouldn't be defined inside methods. This will generate two warnings on every call to `current_user` after the first:

```ruby
(irb):2: warning: already initialized constant YouCannotUseCurrentUserHereCheckTheTerminalOutput
(irb):2: warning: previous definition of YouCannotUseCurrentUserHereCheckTheTerminalOutput was here
```